src/Nox.Solution/Models/Presentation/ApiConfiguration/ApiRouteMappings/ApiRouteMapping.cs

```diff
@@ -74,13 +78,38 @@ public override ValidationResult Validate(NoxSolution topNode, ApiConfiguration
     {
         var result = base.Validate(topNode, parentNode, yamlPath);
 
-        if (HttpVerb == HttpVerb.Get && JsonBodyType != null)
+        if ((HttpVerb is HttpVerb.Get or HttpVerb.Delete) && JsonBodyType != null)
         {
-            result.AddError(nameof(HttpVerb),$"Endpoint [{Name}] with verb [{HttpVerb}] can not define a value for [{nameof(JsonBodyType)}].");
+            result.AddError(nameof(HttpVerb), $"Endpoint [{Name}] with verb [{HttpVerb}] can not define a value for [{nameof(JsonBodyType)}].");
         }
+
+        var parametersFromRoute = ExtractParametersFromRoute();
+        var definedParameters = RequestInput.Select(x => x.Name);
+        foreach (var parameter in parametersFromRoute)
+        {
+            if (!definedParameters.Any(x => string.Equals(x, parameter, StringComparison.InvariantCultureIgnoreCase)))
+            {
+                result.AddError(nameof(RequestInput), $"Endpoint [{Name}] defines a parameter [{parameter}] in the route that is not defined in the [{nameof(RequestInput)}] section.");
+            }
+        }
+
         return result;
     }
 
+    private IEnumerable<string> ExtractParametersFromRoute()
```

Comment:
I would suggest to return an IReadonlyCollection, so its clear that the result is not going to be computed
```suggestion
    private IReadOnlyCollection<string> ExtractParametersFromRoute()
```